index.html

```diff
@@ -0,0 +1,578 @@
+
+### Satisfaction
+
+Given a version `requested` by a document and an `available` version of an implementation, the following pseudocode will determine if the `available` version can satisfy the `requested` version:
+
+```typescript
+canSatisfy(requested: Version, available: Version) =
+  requested.major == available.major &&
+  (requested.major > 0 && (requested.minor <= available.minor)
+    || (requested.minor == available.minor))
+```
+
+### Referencing versions and activating implementations
+
+Schema documents **must** reference a spec version which supports all the schema elements and behaviors required by the document. As a practical matter, authors should prefer to reference the [lowest](#ordering) versions which can [satisfy](#satisfaction) their requirements, as this imposes the fewest constraints on processors and is thus more likely to be supported.
+
+For every spec they support, processors **must** activate an implementation which can [satisfy](#satisfaction) the version required by the document.
+
+### Backus-Naur Form for versions and version tags
+
+```plaintext
+#::[BNF grammar for SemVer specifiers]
+‹version tag› ::= "v" ‹version›
+
+‹version› ::= ‹major›
+            | ‹major› "." ‹minor›
+
+‹major› ::= ‹numeric identifier›
+‹minor› ::= ‹numeric identifier›
+‹numeric identifier› ::= "0"
+                       | ‹positive digit›
+                       | ‹positive digit› ‹digits›
+‹digits› ::= ‹digit›
+           | ‹digit› ‹digits›
+
+‹digit› ::= "0"
+          | ‹positive digit›
+‹positive digit› ::= "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9"
+```
+
+## Extensibility
+
+All directives, scalars, enums, input objects, and other schema elements defined by specs are **extensible**. This means that their definitions within a core schema document do not have to exactly match definitions provided by specs. Instead, definitions should match all usages within the document.
+
+Standard extensibility rules:
+- **Directives** may be defined as taking additional arguments, provided those arguments are [prefixed](#prefixing) with the name of a spec followed by two underscores ("`__`"). These arguments are interpreted as metadata-on-metadata, and are routed to the declaring spec for interpretation
+- **Enums** defined in specs may include values of any name. The space of **unprefixed** names belongs to the document. The space of **prefixed** names belongs to the spec with that prefix, and may be used to pass data between specs.
+- **Input Objects** may be defined with additional fields, provided those fields are [prefixed](#prefixing) with the name of a spec followed by two underscores
+
+
+## Directive Forms
+
+Specs may define as taking multiple **forms**. For example, the `@core` directive itself has two forms:
```

Comment:
I really don't like this whole section. I don't like it as applied to `@core` in specific, and I don't like it as a recommended pattern for other specs to use.

It's actually unclear whether
```
directive @foo on SCHEMA
directive @foo on FIELD_DEFINITION
```

is legal SDL or not! I don't see anything in the GraphQL spec that validates that. However, graphql-js certainly rejects that (see UniqueDirectivesNameRule, and the logic dates way back to graphql/graphql-js#455), which is good enough for me.

Now I get that you know that and that you are suggesting that multiple forms of directives are a spec-specific concept, which get turned into single directives in actual GraphQL. However, I just don't think that's a helpful pattern.

The whole reason that we formally declare directives is to allow tooling to help us write the right directives. Tooling can validate that the correct arguments are passed to directives and can provide auto-complete when you're writing the directives.  Encouraging a pattern where you'll get auto-complete on `exports` for `@core` on SCHEMA when that's not actually part of the same abstract directive just doesn't seem helpful to users working with these things!

I know we can imagine a world where every piece of GraphQL tooling (eg autocomplete) understands not just the directive definition but the spec's machine-parseable multi-form thing, and then the autocomplete will work well. But I think that would be a giant commitment that is an unnecessary scope enlargement.

(I think it could be cool if GraphQL had the feature where directives could have multiple forms with different subsets of arguments and locations.  We could go and propose that to the WG, implement it, etc!  But let's actually do that rather than say "your spec can define things that aren't really how GraphQL works and you have to compile it to a less helpful thing for GraphQL". Remember, "all core schema documents are valid GraphQL schema documents". While I know that's technically true with this proposal, you're writing examples in the spec that aren't actually valid GraphQL because of this feature.)

You've got a great system for specifying multiple different directives with prefixes. I think we should completely drop the concept of directive forms/merging/discernability, and just use different names for different things. So `@core(using: String!, as: String)` and, I dunno, `@core__export(export: Boolean)`